flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/runtime/stream/table/BlackHoleITCase.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.stream.table;
+
+import org.apache.flink.table.factories.BlackHoleTableSinkFactory;
+import org.apache.flink.table.planner.runtime.utils.StreamingTestBase;
+import org.apache.flink.types.Row;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+/**
+ * End to end tests for {@link BlackHoleTableSinkFactory}.
+ */
+public class BlackHoleITCase extends StreamingTestBase {
+
+	@Test
+	public void testTypes() {
+		tEnv().executeSql(
+				"create table blackhole_t (f0 int, f1 double) with ('connector' = 'blackhole')");
+		execInsertTableAndWaitResult(
+				tEnv().fromValues(Arrays.asList(Row.of(1, 1.1), Row.of(2, 2.2))),
```

Comment:
Can be simplified to `tEnv().fromValues(row(1, 1.1), row(2, 2.2)),` use the `org.apache.flink.table.api.Expressions.row`.